sql/core/src/main/scala/org/apache/spark/sql/execution/BufferedRowIterator.java

```diff
@@ -34,6 +34,7 @@
   protected LinkedList<InternalRow> currentRows = new LinkedList<>();
   // used when there is no column in output
   protected UnsafeRow unsafeRow = new UnsafeRow(0);
+  private long startTimeMs = System.currentTimeMillis();
```

Comment:
we should use nanoTime. See https://github.com/databricks/scala-style-guide#misc_currentTimeMillis_vs_nanoTime